### PR TITLE
Download counts: handle several data files

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: app/lib/service/download_counts/sync_download_counts.dart [hunk 1]
```diff
@@ -56,120 +56,157 @@ Map<String, int> _extractDayCounts(Map<String, dynamic> json) {
   return dayCounts;
 }
 
-Future<bool> processDownloadCounts(
-    String downloadCountsFileName, DateTime date) async {
-  final downloadCountsBucket =
+Future<Set<String>> processDownloadCounts(DateTime date) async {
+  final downloadCountsFileName = [
```
Comment (sigurdm) [inline app/lib/service/download_counts/sync_download_counts.dart:60]: Is this still used (except for reporting where we probably should use the actual problematic file name)?

File: app/lib/service/download_counts/sync_download_counts.dart [hunk 2]
```diff
@@ -56,120 +56,157 @@ Map<String, int> _extractDayCounts(Map<String, dynamic> json) {
...
-        return;
-      }
-      final String package;
-      final Map<String, int> dayCounts;
-      try {
-        final data = json.decode(line);
-        if (data is! Map<String, dynamic>) {
-          throw FormatException('Download counts data is not valid json');
-        }
+  await for (final f in bucket.list(prefix: fileNamePrefix)) {
```
Comment (sigurdm) [inline app/lib/service/download_counts/sync_download_counts.dart:80]: Try to only invoke bucket.list once

Maybe do `bucket.list(...).toList();`

File: app/lib/service/download_counts/sync_download_counts.dart (hunk 1 above)
Comment (szakarias) [reply app/lib/service/download_counts/sync_download_counts.dart:60]: nope, removed it.

**Review by sigurdm — APPROVED** [review]

File: app/lib/service/download_counts/sync_download_counts.dart (hunk 2 above)
Comment (szakarias) [reply app/lib/service/download_counts/sync_download_counts.dart:80]: done

**Pull request merged**